### ISSUE TITLE
Error when connecting to Nx Cloud. Code: 404

### ISSUE THREAD

**Issue by zaunermax** [nrwl/ci#65]
Lately we've been getting weird cloud errors where the main job times out with a weird 404 error from nx-cloud. The app shows in the text message that the runs went through, when logging into the cloud the runs also seem to go through but the action itself fails with this error:
![image](https://user-images.githubusercontent.com/15246773/218437014-5b2d1cc0-b145-4785-991e-aa73746eea90.png)

Sadly I've not been able to create a test repo to reproduce this error 🤔 If you need any additional info let me know. 

Our config:
```
name: nrwl-nx-ci

on:
  push:
    branches:
      - main
  pull_request:

jobs:
  main:
    name: Nx Cloud - Main Job
    uses: nrwl/ci/.github/workflows/nx-cloud-main.yml@v0.11.3
    with:
      number-of-agents: 2
      node-version: 18.14.0
      parallel-commands: |
        npx nx-cloud record -- npx nx workspace-lint
        npx nx-cloud record -- npx nx format:check
      parallel-commands-on-agents: |
        npx nx affected --target=lint --parallel=3 --maxWarnings=0
        npx nx affected --target=build --parallel=3 -c prod --prod

  agents:
    name: Nx Cloud - Agents
    uses: nrwl/ci/.github/workflows/nx-cloud-agents.yml@v0.11.3
    with:
      number-of-agents: 2
      node-version: 18.14.0
```

Lastly, is there a way to completely purge the cache of nx cloud? 🤔 I am aware of `--skip-nx-cache` but there is not command afaik that just resets one nx cloud cache?

**Comment by clemenscodes** [comment]
Seems like its a general issue with nx cloud at the moment, happens to me locally and in CI after migrating to 15.7, downgrading to 15.6.3 and nx-cloud 15.0.2 seems to have resolved the issue for now.

**Comment by JamesHenry** [comment]
Hi Folks, sorry about the issues, please could you try with `@nrwl/nx-cloud@15.1.0-beta.3`? @vsavkin will be promoting that to be the latest release later this week I believe

**Comment by clemenscodes** [comment]
No issues after upgrading, appears that version solved it!

**Comment by clemenscodes** [comment]
Unfortunately the error still persists after upgrading in the second CI run and onwards

**Comment by JamesHenry** [comment]
@clemenscodes did you add multiple comments and then delete them again?

**Comment by clemenscodes** [comment]
@JamesHenry yes, I tested it but made a mistake, however now i can confirm that it works with the new version, thanks a lot!

**Comment by JamesHenry** [comment]
Ok great thanks for confirming @clemenscodes. I'll leave this open until that version has been promoted to latests after the US/Canadian holiday later in the week

**Comment by getlarge** [comment]
I hope i won't be polluting this issue (and if yes i can create another issue) but i thought it might be related.
We are migrating from nx 14.5.6 to 15.5.0 and nx-cloud 14.4.0 to 15.0.3 which led to some problem in our CI when trying to connect the agents to Nx cloud with `npx nx-cloud start-agent` (agent job) and `npx nx-cloud start-ci-run` (orchestrator job).
The error :
```
sh: 1: nx-cloud: Permission denied
```

This only happens in for the workflow triggered by the PR that contains our dependencies update. Also i tried to the latest version of nx-cloud `@nrwl/nx-cloud@15.1.0-beta.3`, but the issue remains.
I created some new access token to be sure and also check that our Github app is well configured, now the only reason i can think of is a change in nx-cloud lib or infrastructure. 
@JamesHenry Maybe you have an idea of what happens ?

**Comment by JamesHenry** [comment]
@getlarge This is not related to this issue, I would suggest you double check your setup, because what you are getting looks like a shell level error, not a permissions error related to nx-cloud.

You could reproduce your kind of error in any shell by doing:

```sh
touch nx-cloud
./nx-cloud
```

And you will get `bash: ./nx-cloud: Permission denied` for example in a bash shell. In CI providers, such as Github, it is common for the default shell to show as `sh` so that's your issue.

So make sure you have run an npm/pnpm/yarn install of your dependencies (including `@nrwl/nx-cloud`) before attempting to run commands from it. If the dependency is present, perhaps your node_modules are ending up with wonky permissions somehow. I don't know how you could end up in this state, but again, I really don't think it is anything to do with the nx-cloud library, and definitely not this repo/tooling

**Comment by getlarge** [comment]
I didn't even think that it was a file permission issue as i had the same issue in the CI and locally.
Indeed wiping the (local) node_modules and reinstalling dependencies solved it  and it appears like we are caching the node_modules in our CI, which coulbe be related to those unexecutable executables.
@JamesHenry thanks for your reply and sorry for the noise!

**Comment by raphzandrade** [comment]
Any updates on this issue? Dealing with something similar on my project

**Comment by JamesHenry** [comment]
Hi Folks,

Thanks a lot for your patience on this. A few different (unrelated) issues have been found and fixed which could have caused these 404s.

We have both server-side and client-side fixes which need to be rolled out. Some have already been completed, some will go out over the next week or so once more testing is complete.

Keep an eye out for updates to the `@nrwl/nx-cloud` package and update to the latest as soon as you can.

In the meantime, often forcing a hash change will work around the old issues causing 404s (but sadly not always). So it is always worth trying to force a hash change by e.g. touching a globally depended on file like the `nx.json`

I will close this issue once we have rolled out all fixes for all known causes of 404s.

Many thanks!

**Comment by raphzandrade** [comment]
Updated the package and everything is working fine!

**Comment by meeroslav** [comment]
Closing this issue since it has been resolved.
